background.ts

```diff
@@ -846,6 +848,24 @@ class BackgroundCommandWorker implements CommandWorkerInterface {
     return jsonStringifyWithWhiteSpace(cfg);
   }
 
+  getDiagnosticCategories(): string|undefined {
+    const categories = diagnostics.getCategoryNames();
+
+    return categories && JSON.stringify(categories);
```

Comment:
This shouldn't return a string; it should return the raw object, and have the HTTP layer convert it to JSON instead.  As it is, the API doesn't make much sense (and this is why we had the `text/plain` vs `application/json` confusion).

This also applies to the other new methods.

Reply:
done, although this is the opposite of what the preferences/settings methods are doing